src/ts/util/array.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * @fileoverview Utility module for array sort, binary search.
+ * @author NHN FE Development Lab <dl_javascript@nhn.com>
+ */
+import Schedule from '@src/model/schedule';
+import ScheduleViewModel from '@src/model/scheduleViewModel';
+import { compare } from '@src/time/datetime';
+
+/**
+ * A module for sorting array.
+ * @module array
+ */
+
+/**********
+ * Compare Functions
+ **********/
+
+/**
+ * compare function for array sort.
+ *
+ * sort array by ascending.
+ * @param {boolean} a The boolean to compare
+ * @param {boolean} b The boolean to compare.
+ * @returns {number} Result of comparison.
+ */
+function booleanASC(a: boolean, b: boolean) {
+  if (a !== b) {
+    return a ? -1 : 1;
+  }
+
+  return 0;
+}
+
+/**
+ * compare function for array sort.
+ *
+ * sort array by number ascending.
+ * @param {number} a The number to compare.
+ * @param {number} b The number to compare.
+ * @returns {number} Result of comparison.
+ */
+function numberASC(a: any, b: any) {
+  return Number(a) - Number(b);
+}
+
+/**
+ * compare function for array sort.
+ *
+ * sort array by string ascending
+ * @param {string} _a The string to compare.
+ * @param {string} _b The string to compare.
+ * @returns {number} Result of comparison.
+ */
+function stringASC(_a: any, _b: any) {
+  const a = String(_a),
+    b = String(_b);
+
+  if (a > b) {
+    return 1;
+  }
+  if (a < b) {
+    return -1;
+  }
+
+  return 0;
+}
+
+/**
+ * Compare schedule models for sort.
+ *
+ * 1. all day schedule first.
+ * 2. early start.
+ * 3. longest duration.
+ * 4. early created.
+ * @param {Schedule} a The object schedule instance.
+ * @param {Schedule} b The object schedule instance.
+ * @returns {number} Result of comparison.
+ */
+function scheduleASC(a: Schedule | ScheduleViewModel, b: Schedule | ScheduleViewModel) {
+  const modelA = a instanceof ScheduleViewModel ? a.model : a;
+  const modelB = b instanceof ScheduleViewModel ? b.model : b;
+  const allDayCompare = booleanASC(
+    modelA.isAllDay || modelA.hasMultiDates,
+    modelB.isAllDay || modelB.hasMultiDates
```

Comment:
Property 'hasMultiDates' does not exist on type 'Schedule'.

Reply:
`hasMultipleDates` property has been move from ScheduleViewModel to Schedule.